planner/core/plan_cache_utils.go

```diff
@@ -383,7 +386,7 @@ func (v *PlanCacheValue) MemoryUsage() (sum int64) {
 
 // NewPlanCacheValue creates a SQLCacheValue.
 func NewPlanCacheValue(plan Plan, names []*types.FieldName, srcMap map[*model.TableInfo]bool,
-	matchOpts *utilpc.PlanCacheMatchOpts) *PlanCacheValue {
+	matchOpts *utilpc.PlanCacheMatchOpts, sessVarHints stmtctx.StmtHints) *PlanCacheValue {
```

Comment:
Looks like we should use a pointer here.

Reply:
I clone it because take address may cause the cached stmt hints reset from somewhere unknown.

Reply:
Yes. A clone is needed, `stmtctx` will be shared across statements in a session, therefore `stmtctx.StmtHints` will be shared since it's an embedded struct as a value.